code/modules/projectiles/guns/energy/stun.dm

```diff
@@ -41,6 +41,16 @@
 		tesla_zap(src, 3, 10000)
 	qdel(chain)
 
+/obj/item/weapon/gun/energy/shock_revolver/update_icon()

```

Comment:
Why snowflake this rather than change the icon and names to work with the existing code?